src/compute-client/src/controller.rs

```diff
@@ -1143,7 +1112,12 @@ impl<T: ComputeControllerTimestamp> InstanceState<T> {
             since
         });
 
-        let hold = ReadHold::new(id, since, self.read_holds_tx.clone());
+        let client = self.client.clone();
+        let tx = Box::new(move |id, change| {
+            client.call(move |i| i.report_read_hold_change(id, change));
+            Ok(())
+        });
+        let hold = ReadHold::new(id, since, tx);
```

Comment:
Note that `ReadHold`s issued by the compute controller now contain a copy of `instance::Client` and therefore keep the connected `Instance` alive. I don't see any immediate issue with that, but I might be missing something.